### DISCUSSION TITLE
fourth edition of python machine learning?

### DISCUSSION THREAD

**Discussion by rlacjfjin** [rasbt/machine-learning-book#12] [Q&A] [answered]
Hi,
First of all, thanks for your share with code, I'm very like your python machine learing series.

I have a question:
Is this book considered the fourth edition of python machine learning? Except tensorflow is replaced by pytorch.

Thanks,
Zhe

**Comment by rasbt** [comment, marked as answer]
Thanks for your interest in this! It actually started out as the fourth edition, and if you read the previous edition(s), many contents will be familiar to you. However, there is also a bunch of new stuff besides PyTorch (e.g., two new chapters among others). I have summarized it here in this blog post: [https://sebastianraschka.com/blog/2022/ml-pytorch-book.html](https://sebastianraschka.com/blog/2022/ml-pytorch-book.html)

Let me know in case you have any additional questions.
